File: 0041 first-missing-positive/solution.cs
public class Solution {
    public int FirstMissingPositive(int[] nums) {
       for(int i=0;i<nums.Length;i++) {
            while(nums[i]>0 && nums[i]<nums.Length && nums[nums[i]-1]!=nums[i]) {
                int temp=nums[i];
                nums[i]=nums[temp-1];
                nums[temp-1]=temp;
            }
       }
        for(int i=0;i<nums.Length;i++) {
            if(nums[i]!=i+1) return i+1;
        }
        return nums.Length+1;
    }
}

File: 0041 first-missing-positive/solution.rs
impl Solution {
    pub fn first_missing_positive(mut nums: Vec<i32>) -> i32 {
        for i in 0..nums.len() {
            while nums[i] > 0 
            && (nums[i] as usize) <= nums.len() 
            && nums[i] != nums[(nums[i] - 1) as usize] 
            {
                let idx = (nums[i] -1) as usize;
                nums.swap(i,idx);
            }
        }
        for i in 0..nums.len() {
            if (i as i32) + 1 != nums[i] {
                return (i as i32) + 1;
            }
        }   
        return (nums.len() as i32) +1
    }
}
